### PR TITLE
converted the old tailwind css example to typescript 

### PR DESCRIPTION
**Pull request by kumard3**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: examples/with-tailwindcss/package.json [hunk 1]
```diff
@@ -6,13 +6,18 @@
     "start": "next start"
   },
   "dependencies": {
-    "next": "latest",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "next": "12.0.7",
```
Comment (balazsorban44) [inline examples/with-tailwindcss/package.json:9]:
```suggestion
    "next": "latest",
```

File: examples/with-tailwindcss/package.json [hunk 2]
```diff
@@ -6,13 +6,18 @@
...
+    "react": "17.0.2",
+    "react-dom": "17.0.2"
```
Comment (balazsorban44) [inline examples/with-tailwindcss/package.json:11]:
```suggestion
    "next": "latest",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
```

I don't see a reason why this should change.

File: examples/with-tailwindcss/tailwind.config.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-    './components/**/*.{js,ts,jsx,tsx}',
+    "./pages/**/*.{js,ts,jsx,tsx}",
+    "./components/**/*.{js,ts,jsx,tsx}",
   ],
   theme: {
     extend: {},
   },
   plugins: [],
-}
+}
```
Comment (balazsorban44) [inline examples/with-tailwindcss/tailwind.config.js:10]: This file should have been untouched.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks.

I left some comments.

Could you also please run `npx prettier --write ./examples/with-tailwindcss` to fix the lint errors and commit the changes?

**Comment by kumard3** [comment]
> Thanks.
> 
> I left some comments.
> 
> Could you also please run `npx prettier --write ./examples/with-tailwindcss` to fix the lint errors and commit the changes?

Yes sure will be doing that

**Comment by balazsorban44** [comment]
Please let's focus this PR on the tailwind example. If you would like to convert other examples to TypeScript, you are welcome to do so, but open a separate PR instead. It makes it easier to track down changes in the future. Thank you.

**Comment by kumard3** [comment]
> Please let's focus this PR on the tailwind example. If you would like to convert other examples to TypeScript, you are welcome to do so, but open a separate PR instead. It makes it easier to track down changes in the future. Thank you.

Sure will we doing that, and from on I will be doing this in a systemic way.

File: examples/progressive-web-app/pages/api/hello.js [hunk 4]
```diff
@@ -4,4 +4,4 @@ const hello = (req, res) => {
   res.status(200).json({ name: 'John Doe' })
 }
 
-export default hello
+export default hello
```
Comment (leerob) [inline examples/progressive-web-app/pages/api/hello.js:7]: It appears there are unrelated changes to files other than the Tailwind example - could we revert all of those?

Comment (kumard3) [reply examples/progressive-web-app/pages/api/hello.js:7]: Yes sure

**Comment by kumard3** [comment]
@leerob I have  reverted all the unrelated files

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Please remove **all** the changes, except in the folder: `examples/with-tailwindcss` as requested here https://github.com/vercel/next.js/pull/32808#issuecomment-1001258194 (I could have been more explicit)

There are still changes within `examples/progressive-web-app`

**Comment by kumard3** [comment]
Can you please guide me how to do so, because this is my first contribution.

> Please remove **all** the changes, except in the folder: `examples/with-tailwindcss` as requested here [#32808 (comment)](https://github.com/vercel/next.js/pull/32808#issuecomment-1001258194) (I could have been more explicit)
> 
> There are still changes within `examples/progressive-web-app`

Can you please guide me how to do so, because this is my first contribution.

**Comment by kumard3** [comment]
> Please remove **all** the changes, except in the folder: `examples/with-tailwindcss` as requested here [#32808 (comment)](https://github.com/vercel/next.js/pull/32808#issuecomment-1001258194) (I could have been more explicit)
> 
> There are still changes within `examples/progressive-web-app`

I have  reverted the commit , but it is still showing the same changes in file, can you please guide me , to performer this task.

**Comment by kumard3** [comment]
@balazsorban44 I have reverted all the unrelated files .

**Review by balazsorban44 — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

File: examples/with-tailwindcss/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
```
Comment (leerob) [inline examples/with-tailwindcss/.eslintrc.json:2]: I don't think we should include ESLint - while it is helpful, I think this example should be as simple as possible - just Next.js and Tailwind.

File: examples/with-tailwindcss/README.md [hunk 6]
```diff
@@ -1,3 +1,11 @@
+<p  align="center">
+
+<img  src="https://res.cloudinary.com/ddcg0rzlo/image/upload/v1640340715/nextjs-tailwind-typescript-banner_vslgq4.png"  alt="Next.js TypeScript Starter">
```
Comment (leerob) [inline examples/with-tailwindcss/README.md:3]: Could we remove this image please?

**Comment by balazsorban44** [comment]
@kumard3 could you please follow up in a separate PR with @leerob's suggestion?

**Comment by LazirusDon** [comment]
example is typescript now?  how do i start a nextjs tailwind without typescript ?
how to initialize next/tailwind with just js?

npx create-next-app --example with-tailwindcss with-tailwindcss-app
converted the old tailwind css example to typescript (#32808),

**Comment by leerob** [comment]
Yes, the example is now using TypeScript. If you would prefer not to use it, you can remove the TS configuration (it's a few files/lines).

**Comment by LazirusDon** [comment]
Thanks for the reply dude, I’m new at programming I figured it out just a few changes a little clean up
Thanks again dude

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by ahmadbilaldev** [comment]
Why was the JavaScript example changed to TypeScript? I think the default example should be using JavaScript.

We can do something like `-with-ts-tailwindcss` for the TypeScript instead.

**Comment by msaaddev** [comment]
@ahmadbilaldev totally agrees with you here. There should also be a JavaScript-based template available for the Next.js TailwindCSS example.

**Comment by balazsorban44** [comment]
The way forward is that we are going to focus on TypeScript and prefer it over JS, as it provides more safety even for beginner developers, and is being more and more widely adopted. 

We are discussing alternatives like a `--js` flag that could potentially harness SWC to automatically convert examples to plain JavaScript, but nothing is set yet. It is easier to go from TypeScript to JavaScript, not the other way around.
